jspoon/src/main/java/pl/droidsonroids/jspoon/FieldType.java

```diff
@@ -0,0 +1,286 @@
+package pl.droidsonroids.jspoon;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Array;
+import java.lang.reflect.Field;
+import java.lang.reflect.GenericArrayType;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.lang.reflect.TypeVariable;
+import java.lang.reflect.WildcardType;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FieldType {
```

Comment:
It shouldn't be public

Reply:
I think `ElementConverter` interface of custom converters should include `FieldType`, `SelectorSpec` and `Element` (in this order), because currently its very limited since it hasno access to field's metadata, a custom converter cannot have implementation based on field's type or look for any custom field annotations. For example you can't implement a `CustomStringConverter extends ElementConverter<String>` that performs validation if there is a custom `@ValideNumbersOnly` annotation on the field).
I intentionally didn't include such change in this PR because it's backwards incompatible. But in the next PR I want to make changes to librarie's exceptions which is anther backwards incompatible change so everything what is backwards incompatible would be under one PR, including removal of now deprecated classes/values. 

Having said all that, I can change this to package-private but it will have to be public in with next PR.

Reply:
It should have javadocs if it is public.

Reply:
Good point. I'll add that to both this and SelectorSpec

Reply:
> Having said all that, I can change this to package-private but it will have to be public in with next PR.

@cr3ativ3 Are you going to work on this?